web/src/helpers/cadPackages/openScadController.js

```diff
@@ -13,38 +13,52 @@ export const render = async ({ code, settings }) => {
     },
     file: code,
   })
-  const response = await fetch(openScadBaseURL + '/render', {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body,
-  })
-  if (response.status === 400) {
-    const { error } = await response.json()
-    const cleanedErrorMessage = error.replace(
-      /["|']\/tmp\/.+\/main.scad["|']/g,
-      "'main.scad'"
-    )
+  try {
+    const response = await fetch(openScadBaseURL + '/render', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body,
+    })
+    if (response.status === 400) {
+      const { error } = await response.json()
+      const cleanedErrorMessage = error.replace(
+        /["|']\/tmp\/.+\/main.scad["|']/g,
+        "'main.scad'"
+      )
+      return {
+        isError: true,
+        message: {
+          type: 'error',
+          message: cleanedErrorMessage,
+        },
+      }
+    }
+    const data = await response.blob()
+    return {
+      objectData: {
+        type: 'png',
+        data,
+      },
+      message: {
+        type: 'message',
+        message: 'successful render',
+      },
+    }
+  } catch (e) {
```

Comment:
Most of this is white space change from indentation, adding a try catch block, unrelated to the ide viewer :/